### DISCUSSION TITLE
Why `python-alpha` & `python-rc`?

### DISCUSSION THREAD

**Discussion by YSCohen** [ScoopInstaller/Versions#727] [General]
There's only one Python pre at a time

**Comment by rasa** [comment]
Good point. It seems the checkver could be made smart enough to capture both:
https://github.com/ScoopInstaller/Versions/blob/4a96e4695c8d23c52c10033703b38cb7fae8747b/bucket/python-alpha.json#L102-L102
https://github.com/ScoopInstaller/Versions/blob/4a96e4695c8d23c52c10033703b38cb7fae8747b/bucket/python-rc.json#L102

**Reply by niheaven** [reply]
Add some backgrounds: these two manifests were originally just for testing, to keep `Versions` bucket live. 🤣 

Will migrate to one regex and add a `python-beta` just likes `python39-beta`.

**Reply by YSCohen** [reply]
Huh? why are you adding `python-beta`? why not just consolidate all 3 (`alpha`, `beta`, `rc`) into `python-pre`?

**Reply by niheaven** [reply]
Scoop does not have manifest migrated feature, so old ones will be kept. `beta` only means it is in beta stage.

**Reply by YSCohen** [reply]
that just exacerbates the problem that i'm trying to emphasize here

**Reply by rasa** [reply]
@YSCohen: @niheaven implemented your suggestion, but named it `beta` instead of `pre`. It includes all alpha, beta, and rc releases. In a perfect world, for those who have already installed the old alpha, or rc versions, they would now "point" to this new version. But that functionality does not exist... yet. So we leave the old versions alone, to not annoy users when their manifests disappear without warning.

**Reply by YSCohen** [reply]
ok. may as well name it `python-pre`, because otherwise the existence of `alpha` and `rc` will make it seem like this is literally _beta_

**Reply by niheaven** [reply]
Okay, another `python-pre` will be added (https://github.com/ScoopInstaller/Versions/commit/dac5cdae450be744c269d33aa6c80d13285a42b0), and I'll try to implement the migrating or pointing feature (as brew's).

**Reply by YSCohen** [reply]
![image](https://user-images.githubusercontent.com/78832113/198497682-0f7c3b39-b171-44a1-bd41-7a9199a9242a.png)
pythons proliferate

**Reply by rasa** [reply]
> pythons proliferate

It's the Versions bucket, so it's kinda expected, I guess. It will get cleaned up eventually. Thanks for raising the issue.
